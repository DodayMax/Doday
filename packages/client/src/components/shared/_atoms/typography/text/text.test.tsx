import * as React from 'react';
import { shallow } from 'enzyme';
import { Text } from '@shared';

import {
  TypographySize,
  TypographyColor,
  TypographyAlignment,
} from '@lib/common-interfaces';

describe('Text', () => {
  it('renders correctly without props', () => {
    const text = shallow(<Text>test</Text>);

    expect(text).toMatchSnapshot();
    expect(text.find('span').contains('test')).toBe(true);
    expect(text.find('span').hasClass('text-l')).toBe(true);
  });

  it('renders correctly with props', () => {
    const text = shallow(
      <Text
        size={TypographySize.s}
        color={TypographyColor.Primary}
        align={TypographyAlignment.Center}
      >
        test
      </Text>
    );

    expect(text.find('span').contains('test')).toBe(true);
    expect(text.find('span').hasClass('text-s')).toBe(true);
    expect(text.find('span').hasClass('primary-text')).toBe(true);
    expect(text.find('span').hasClass('align-center')).toBe(true);
  });
});
