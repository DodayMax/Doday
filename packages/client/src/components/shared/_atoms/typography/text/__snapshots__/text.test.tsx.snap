// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown>
    test
  </Unknown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test",
      "className": "text-l",
    },
    "ref": null,
    "rendered": "test",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
        "className": "text-l",
      },
      "ref": null,
      "rendered": "test",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
